walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +272,155 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void noTimeSlotsWithOptionalAtendee() {
+    // Optinal atendee has an all-day event. 
+    // We should see the two options that fit the mandatory attendees, ignoring the optional atendee,
+    // because there is no slots available for both mandatory and optional attendees.
+    //
+    // Events of mandetory attendees  :        |--A--|     |--B--|
+    // Events of optional attendee    :  |-------------C---------------|
+    // Day                            :  |-----------------------------|
+    // Options to return              :  |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.WHOLE_DAY, Arrays.asList(PERSON_C)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+        request.addOptionalAttendee(PERSON_C);
```

Comment:
You can have this inline with the previous line (same for the rest).

Reply:
It also looked wierd to me that it is in a new line when I wrote it, but because addOptionalAttendee returns void I get a compilation error because it is supposed to be a MeetingRequest type. How would you do this?

Reply:
Ah I see, fine in that case (nothing you can do).